crates/engine/tree/src/tree/mod.rs

```diff
@@ -1391,11 +1391,14 @@ where
                 }
                 Err(err) => {
                     debug!(target: "engine", ?err, "failed to connect buffered block to tree");
+                    if let Err(fatal) = self.on_insert_block_error(err) {
+                        warn!(target: "engine", ?fatal, "fatal error occurred while connecting buffered blocks");
```

Comment:
fatal is mislead here imo because fatal means that this block is invalid not that a fatal error happened?
could we rephrase this?

Reply:
Validation errors should be filtered out and handled in `insert_block_error`, the type in the `Err` is an `InsertBlockFatalError`

Reply:
yeah, my point is how should "fatal" be interpreted if a user sees this in the terminal

> fatal error occurred while connecting buffered blocks

Reply:
because we don't terminate the task here, right? so it's not fatal, but maybe it should?

Reply:
> because we don't terminate the task here, right? so it's not fatal, but maybe it should?

Yeah, I guess we should bubble these up, since it would be an error that we can't otherwise do anything with

Reply:
okay, no I fully understood it, this is indeed a fatal internal error but can don't have an easy way to terminate the service, since we don't really expect this case under normal ops, this is appropriate until we find a way to either terminate the task or emit an error message